libs/transactions/deposit.ts

```diff
@@ -1,130 +1,49 @@
-import Queue from "queue";
-import Web3 from "web3";
+import type Web3 from "web3";
 import path from "path";
 import { parentLogger } from "../utils";
-import { Token } from "../tokens";
-import { submitTransaction } from "./transactions";
-import { toBaseUnit } from "./units";
+import type { Token } from "../tokens";
+import { submitTransaction } from "./helpers/submit";
+import type { Client } from "../client";
+import type { NightfallZkpKeys } from "../nightfall/types";
+import type { TransactionReceipt } from "web3-core";
 
 const logger = parentLogger.child({
   name: path.relative(process.cwd(), __filename),
 });
 
-// TODO improve return type
-async function setToken(
-  tokenAddress: string,
-  tokenStandard: string,
+export async function createAndSubmitDeposit(
+  token: Token,
+  ownerAddress: string,
+  ownerPrivateKey: string,
+  ownerZkpKeys: NightfallZkpKeys,
+  shieldContractAddress: string,
+  value: string,
+  fee: string,
   web3: Web3,
-): Promise<any> {
-  logger.debug({ tokenAddress, tokenStandard }, "setToken");
+  client: Client,
+) {
+  logger.debug("createAndSubmitDeposit");
 
-  // TODO validate and format tokenAddress, tokenStandard
-  const token = new Token({
-    web3: web3,
-    address: tokenAddress,
-    standard: tokenStandard,
-  });
+  const resData = await client.deposit(token, ownerZkpKeys, value, fee);
+  // resData null signals that something went wrong in the Client
+  if (resData === null) return;
```

Comment:
We could improve this, if nothing is reported the user might think everything worked well.

Reply:
TBD error handling